database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala

```diff
@@ -82,10 +84,14 @@ trait WorkflowMetadataSummaryEntryComponent {
       val workflowStatusFilter = workflowStatuses.
         map(workflowStatus => workflowMetadataSummaryEntry.workflowStatus.fold(ifEmpty = exclude)(_ === workflowStatus)).
         reduceLeftOption(_ || _)
-
+      // Workflows can be queried by multiple labels, the reduceLeftOption ANDs together all workflows are associated
+      // to all the labels from the query.
+      val labelsFilter = labelKeyLabelValues.map { case (labelKey, labelValue) =>
+          existsWorkflowIdLabelKeyAndValue(workflowMetadataSummaryEntry.workflowExecutionUuid, labelKey, labelValue) }.
```

Comment:
ToL (I know you've thought about this way more than me!): This seems like a lot of work going on in Cromwell that could be compiled into a database query?